contracts/BufferedToken.sol

```diff
@@ -0,0 +1,109 @@
+pragma solidity ^0.6.7;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "./WadMath.sol";
+import "./oracles/IOracle.sol";
+
+/**
+ * @title Buffered Token
+ * @author Alex Roan (@alexroan)
+ * @notice This contract represents a token which accepts ETH as collateral to mint and burn ERC20 tokens.
+ * It's functions abstract all oracle, conversion, debt ratio and ETH buffer calculations, so that
+ * inheriting contracts can focus on their minting and burning algorithms.
+ */
+contract BufferedToken is ERC20 {
+    using SafeMath for uint;
+    using WadMath for uint;
+
+    uint constant WAD = 10 ** 18;
+
+    uint public ethPool = 0;
```

Comment:
Is the `ethPool` variable redundant with `address(this).balance`, ie, the actual amount of ETH currently stored by the contract?  Maybe we should just use the latter instead, ie, define a function `ethPool()` that just returns `address(this).balance`?

Reply:
That will be the case once we merge `BuffereToken.sol` and `USM.sol` into one.

Right now, it can't be guaranteed that `eth` actually means ether, it could be any random ERC20. Which could be a useful feature to keep!

Reply:
I see.  Yeah, I don't see us using anything other than ETH (or WETH?) anytime soon but I see your point in terms of the design.